engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.cpp

```diff
@@ -86,16 +86,21 @@ SortAndSampleKernel::SortAndSampleKernel(std::size_t kernel_id, const std::strin
     this->query_graph = query_graph;
     set_number_of_message_trackers(2); //default
     this->output_.add_port("output_a", "output_b");
+    get_samples = true;
+    already_computed_partition_plan = false;
 }
 
-void SortAndSampleKernel::compute_partition_plan(std::vector<ral::frame::BlazingTableView> sampledTableViews, std::size_t avg_bytes_per_row, std::size_t local_total_num_rows) {
+void SortAndSampleKernel::compute_partition_plan(
+    std::vector<std::unique_ptr<ral::frame::BlazingTable>> inputSamples,
```

Comment:
why did we switch from views to tables again?

Reply:
This functions is now effectively a task, therefore it needs to operate on Tables not views